cluster/gce/upgrade.sh

```diff
@@ -464,13 +464,8 @@ function update-coredns-config() {
   cleanup() {
     rm -rf "${download_dir}"
   }
-  trap cleanup EXIT
+  trap cleanup RETURN
 
-  # Get the new installed CoreDNS version
-  echo "Waiting for CoreDNS to update"
```

Comment:
this was added in https://github.com/kubernetes/kubernetes/pull/78302, want an ack from @MrHohn before removing it

Reply:
Trying to understand - why is removing this step necessary for fixing the upgrade test?

cc @rajansandeep

Reply:
The check for `resourceVersion` is added for the detection of changes when CoreDNS is updating. 
But if there is no change in CoreDNS (eg: upgrading from 1.19 to 1.19), it might end up getting stuck here.

Reply:
Thanks for the explanation. Does it mean we may get the old coredns version after removing this waiting step? Or if this is not concerning in practice then it looks fine to me.

Reply:
I can't tell if the upgrade/downgrade logic is load-bearing... are people using this for actual upgrades or is this just for tests?

if the config upgrade/downgrade is required to work, I'd expect that to be done in a more prescriptive way, like the etcd `cluster/images/etcd/migrate-if-needed.sh` approach, rather than here

Reply:
> if the config upgrade/downgrade is required to work, I'd expect that to be done in a more prescriptive way, like the etcd cluster/images/etcd/migrate-if-needed.sh approach, rather than here

Agreed. My understanding is that this script is mostly used for tests but not in production at the moment.

Reply:
Another thought - instead of removing this step completely, could we set a max wait time here? If the version doesn't change for X minutes, we continue regardless. It may slow down the upgrade a bit but keeps the coredns version upgrade logic intact.

cc @mtaufen

Reply:
Thanks @MrHohn. Ideally we also make it configurable. What's a sane default value?

Reply:
Looking through the steps, the previous step `upgrade-master()` would wait for the new master to respond to API requests - then 3 minutes sounds more than enough. Alternatively we can make the step of waiting for coredns version updated configurable so tests can set it on-demand.

Reply:
the loop will timeout in 3 minutes now. PTAL